tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EncryptedEmailFastViewGetMethod.scala

```diff
@@ -0,0 +1,128 @@
+                              mailboxSession: MailboxSession): SMono[EncryptedEmailGetResponse] =
+    computeResults(request, mailboxSession)
+      .map(result => EncryptedEmailGetResponse(
+        accountId = request.accountId,
+        state = UuidState.INSTANCE,
+        list = result.list,
+        notFound = result.notFound
+      ))
+
+  private def computeResults(request: EncryptedEmailGetRequest,
+                             mailboxSession: MailboxSession): SMono[EncryptedEmailFastViewResults] = {
+    val parsedIds: List[Either[(UnparsedEmailId, IllegalArgumentException), MessageId]] = request.ids.value
+      .map(asMessageId)
+
+    val messagesIds: List[MessageId] = parsedIds.flatMap({
+      case Left(_) => None
+      case Right(messageId) => Some(messageId)
+    })
+
+    val parsingErrors: SFlux[EncryptedEmailFastViewResults] = SFlux.fromIterable(parsedIds.flatMap({
+      case Left((id, _)) =>
+        Some(EncryptedEmailFastViewResults.notFound(id))
+      case Right(_) => None
+    }))
+
+    SFlux.merge(Seq(parsingErrors, retrieveEmails(messagesIds, mailboxSession)))
+      .reduce(EncryptedEmailFastViewResults.empty())(EncryptedEmailFastViewResults.merge)
+  }
+
+  private def retrieveEmails(ids: Seq[MessageId], mailboxSession: MailboxSession): SFlux[EncryptedEmailFastViewResults] =
+    SFlux.fromIterable(ids)
+      .flatMap(messageId => retrieveEmail(messageId, mailboxSession))
+
+  private def retrieveEmail(ids: MessageId, mailboxSession: MailboxSession): SMono[EncryptedEmailFastViewResults] =
+    encryptedEmailFastViewReader.read(ids, mailboxSession)
+      .map(messageIdAndFastView => EncryptedEmailFastViewResults.list(messageIdAndFastView._1, messageIdAndFastView._2))
+      .onErrorResume {
+        case messageNotFound: MessageNotFoundException => SMono.just(EncryptedEmailFastViewResults.notFound(messageNotFound.messageId))
+        case messageNotFound: com.linagora.tmail.encrypted.MessageNotFoundException => SMono.just(EncryptedEmailFastViewResults.notFound(messageNotFound.messageId))
+        case error => SMono.error(error)
+      }
+
+  private def asMessageId(id: UnparsedEmailId): Either[(UnparsedEmailId, IllegalArgumentException), MessageId] =
+    try {
+      Right(messageIdFactory.fromString(id.id))
+    } catch {
+      case e: Exception => Left((id, new IllegalArgumentException(e)))
+    }
+}
+
+private class EncryptedEmailFastViewReader @Inject()(encryptedEmailContentStore: EncryptedEmailContentStore,
+                                                     messageIdManager: MessageIdManager) extends EncryptedEmailViewReader[(MessageId, EncryptedEmailFastView)] {
+  override def read[T >: (MessageId, EncryptedEmailFastView)](messageId: MessageId, mailboxSession: MailboxSession): SMono[T] =
+    isMessageExits(messageId, mailboxSession)
+      .filter(isMessageExits => isMessageExits)
+      .flatMap(_ => SMono.fromPublisher(encryptedEmailContentStore.retrieveFastView(messageId)))
```

Comment:
On such flatMaps we likely should limit the concurrency...

```suggestion
      .flatMap(_ => SMono.fromPublisher(encryptedEmailContentStore.retrieveFastView(messageId), DEFAULT_CONCURRENCY))
```